src/lib/classes/pdl/ProductIndex.class.php

```
@@ -1605,5 +1616,369 @@ public function getConnection() {
+        $params[] = $query->minLongitude;
+        $params[] = $query->maxLongitude;
+      }
+    } elseif (isset($query->maxLongitude)) {
+      //If only set a max on longitude
+      if ($query->maxLongitude > 180) $query->maxLongitude -= 360;
+      $where[] = 'ps.' . self::SUMMARY_EVENT_LONGITUDE . '<=?';
+      $params[] = $query->maxLongitude;
+    } elseif (isset($query->minLongitude)) {
+      //If only set a min on longitude
+      if ($query->minLongitude < -180) $query->minLongitude += 360;
+      $where[] = 'ps.' . self::SUMMARY_EVENT_LONGITUDE . '>=?';
+      $params[] = $query->minLongitude;
+    }
+
+    $sql .= 'WHERE ' . implode(' AND ', $where);
+    if ($getProperties) $propertySql .= ' WHERE ' . implode(' AND ', $where);
+
+    if ($getProperties) return array($sql,$propertySql,$params);
+    return array($sql, $params);
+
+  }
+
+  /**
+   * Gets number of products returned by product query
+   * 
+   * @param $query {ProductQuery}
+   *    ProductQuery storing supplied information
+   * 
+   * @return {int}
+   *    Number of products returned by query
+   */
+  public function getProductCount($query) {
+    $search = $this->buildProductSearchSql($query);
+    
+    $sql = preg_replace("/SELECT(.*?)FROM/s", "SELECT COUNT(*) FROM", $search[0]); //Maybe update getCount... so i'm not copying code here
+    $statement = $this->connection->prepare($sql);
+    $statement->execute($search[1]);
+
+    $count = intval($statement->fetch()[0]);
+
+    return $count;
+  }
+
+  /**
+   * Gets products from query
+   * 
+   * @param $query {ProductQuery}
+   *    ProductQuery storing supplied information
+   */
+  public function getProductSummaryArray($query) {
+    $search = $this->buildProductSearchSql($query, true);
+
+    //Execute product sql statement
+    $productStatement = $this->connection->prepare($search[0]);
+    if ($productStatement->execute($search[2]) == false) {
+      print_r($productStatement->errorInfo());
```

throw an exception instead of printing the error.